options/options.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/caarlos0/env/v6"
 
 	"github.com/ossf/scorecard/v4/clients"
+	"github.com/ossf/scorecard/v4/clients/githubrepo"
```

Comment:
That sounds like a good idea, since this effects alot of code, is it ok to make modifications to existing tests to account for the commit depth being passed to InitRepo?

Reply:
Yes, for sure. We could have the behavior such that when `commit-depth`/`number-of-commits` is `0` the RepoClient implementation uses some implementation specific default value (30 in case of `githubrepo`). That way, you'd only have to update the existing tests with `InitRepo(..., /*number-of-commits*/ 0)` and the behavior of the tests should remain the same.

Reply:
@azeemshaikh38 i just made changes to the InitRepo Function to accept a number of commits, i also added a getcommitdepth func to the interface as i was did not want to pass the arg to pkg.Scorecards func that runs the checks. let me know if this makes sense. some functions in the github client were modified to accept a commit depth var as well.

Reply:
Thanks! I would prefer not having getcommitdepth func in the interface, but not a blocker to merge this PR. Would appreciate if you'd remove the func, but feel free to leave it as is and we can clean up later.

Reply:
@azeemshaikh38 no problem i will remove it and all instances of 30 will be 0 and handled in InitRepo. will fix tests to reflect that as well.